src/compiler/checker.ts

```diff
@@ -1020,38 +1020,68 @@ namespace ts {
             return links.resolvedExports || (links.resolvedExports = getExportsForModule(moduleSymbol));
         }
 
-        function extendExportSymbols(target: SymbolTable, source: SymbolTable) {
+        interface ExportStarDiagnosticsLookupTable {

```

Comment:
Why not just make this a `Map<ExportDeclaration[]>` and use the `id` as the text?